include/circt/Dialect/LTL/LTLOps.td

```diff
@@ -176,6 +185,52 @@ def RepeatOp : LTLOp<"repeat", [Pure]> {
   }];
 }
 
+def GoToRepeatOp : LTLOp<"goto_repeat", [Pure]> {
```

Comment:
It's probably worth checking that `base` and `more` actually are >= 0. You can probably add this as an op trait:
```suggestion
def GoToRepeatOp : LTLOp<"goto_repeat", [
  Pure,
  PredOpTrait<"'base' must be >= 0", CPred<"$base >= 0">>,
  PredOpTrait<"'more' must be >= 0", CPred<"$more >= 0">>
]> {
```